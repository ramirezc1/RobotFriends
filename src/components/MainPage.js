import React from "react";

import CardList from "../components/CardList";
import Scroll from "../components/Scroll";
import SearchBox from "../components/SearchBox";
import ErrorBoundary from "../components/ErrorBoundary";

class MainPage extends React.Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }
  filterRobots = () => {
    return this.props.robots.filter((robot) => {
      return robot.name
        .toLowerCase()
        .includes(this.props.searchField.toLowerCase());
    });
  };
  render() {
    const { onSearchChange, isPending } = this.props;

    return (
      <div className="tc">
        <h1 className="f1">RobotFriends</h1>
        <SearchBox onSearchChange={onSearchChange}></SearchBox>
        <Scroll>
          {isPending ? (
            <h1>Loading...</h1>
          ) : (
            <ErrorBoundary>
              <CardList robots={this.filterRobots()}></CardList>
            </ErrorBoundary>
          )}
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
