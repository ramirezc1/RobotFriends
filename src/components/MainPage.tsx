import React from "react";

import CardList from "./CardList";
import Scroll from "./Scroll";
import SearchBox from "./SearchBox";
import ErrorBoundary from "./ErrorBoundary";
export interface IRobot{
  name:string,
  id:number,
  email:string,
}

interface IRobotProps{
  onRequestRobots?:void
  onSearchChange?:void
}
class MainPage extends React.Component <IRobotProps, IRobot> {
  componentDidMount() {
    // this.props.onRequestRobots();
  }
  filterRobots = () => {
    // return this.props.robots.filter((robot) => {
    //   return robot.name
    //     .toLowerCase()
    //     .includes(this.props.searchField.toLowerCase());
    // });
  };
  render() {
    const { onSearchChange:any, isPending:any } = this.props;

    return (
      <div className="tc">
        <h1 className="f1">RobotFriends</h1>
        <SearchBox onSearchChange={onSearchChange}></SearchBox>
        <Scroll>
          {isPending ? (
            <h1>Loading...</h1>
          ) : (
            <ErrorBoundary>
              <CardList robots={this.filterRobots()}></CardList>
            </ErrorBoundary>
          )}
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
