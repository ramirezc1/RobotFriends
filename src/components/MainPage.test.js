import { shallow } from "enzyme";
import React from "react";

import MainPage from "./MainPage";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "",
    isPending: false,
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

// it("renders Main page with out crashing", () => {
//   expect(wrapper).toMatchSnapshot();
// });

//183 udemy
it("it filters robots correctly", () => {
  expect(wrapper.instance().filterRobots([])).toEqual([]);
});
