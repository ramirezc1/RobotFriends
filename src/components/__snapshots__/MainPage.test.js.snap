// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it filters robots correctly 1`] = `undefined`;

exports[`renders Main page with out crashing 1`] = `undefined`;

exports[`renders Main page with out crashing 2`] = `undefined`;
